### PR TITLE
Provide helpful stack trace in WritableResourceStream

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

**Comment by trowski** [comment]
Personally I would pass a closure or null to the suspension to resume, e.g. `static fn () => throw new ClosedException("The stream was closed by the peer");`. It adds another function call to the stack trace, but the code seems less strange and Psalm should be more accepting of it.

File: src/WritableResourceStream.php [hunk 1]
```diff
@@ -128,26 +140,6 @@ static function ($callbackId, $stream) use (
...
-
-                    if (\is_resource($resource)) {
-                        $meta = \stream_get_meta_data($resource);
-                        if (\str_contains($meta["mode"], "+")) {
-                            \stream_socket_shutdown($resource, \STREAM_SHUT_WR);
-                        } else {
-                            \fclose($resource);
-                        }
-                        $resource = null;
-                    }
```
Comment (trowski) [inline src/WritableResourceStream.php:150]: We still need to do these things, no?

Comment (kelunik) [reply src/WritableResourceStream.php:150]: It's mostly done in the `finally` block, but I guess we need to keep the loop? Currently it repeatedly invokes the watcher, as it's not disabled in that case.

Comment (trowski) [reply src/WritableResourceStream.php:150]: Maybe repeatedly invoking the watcher is better since it will provide a better backtrace.

Comment (trowski) [reply src/WritableResourceStream.php:150]: I didn't see this code was duplicated in the finally block – oops. Yeah, looks like this is no longer necessary.

**Pull request merged**